packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,197 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/foundation.dart';
+
+import 'package:flutter/cupertino.dart';
+
+const EdgeInsets _kHorizontalItemPadding =
+    const EdgeInsets.symmetric(horizontal: 16.0, vertical: 0.0);
+
+const Radius defaultCornerRadius = const Radius.circular(3.0);
+
+/// An iOS-style segmented control bar
+///
+/// Displays the provided [options] widgets in a horizontal list.
+/// Exactly one [options] widget is always selected. By default,
+/// the leftmost widget is selected.
+///
+/// All widgets have the same width.
+///
+/// All widgets provided should have unique Key property.
+/// These keys will be used to identify the currently selected
+/// widget.
+///
+/// See also:
+///
+/// * <https://developer.apple.com/design/human-interface-guidelines/ios/controls/segmented-controls/>
+class SegmentedControl extends StatefulWidget {
+  /// Creates an iOS-style segmented control bar.
+  const SegmentedControl(
```

Comment:
I'm curious how the a widget that embeds this learns of the change in selection: I was expecting to see an `onValueChanged` argument here that is a `ValueChanged<Key>`, and a `Key value` that is the key of the currently selected option widget.